lib/StencilConfigManager.js

```diff
@@ -0,0 +1,134 @@
+require('colors');
+const fsModule = require('fs');
+const path = require('path');
+
+const fsUtilsModule = require('./utils/fsUtils');
+const { THEME_PATH } = require('../constants');
+
+class StencilConfigManager {
+    constructor({
+        themePath = THEME_PATH,
+        fs = fsModule,
+        fsUtils = fsUtilsModule,
+        logger = console,
+    } = {}) {
+        this.oldConfigFileName = '.stencil';
+        this.configFileName = 'config.stencil.json';
+        this.secretsFileName = 'secrets.stencil.json';
+
+        this.themePath = themePath;
+        this.oldConfigPath = path.join(themePath, this.oldConfigFileName);
+        this.configPath = path.join(themePath, this.configFileName);
+        this.secretsPath = path.join(themePath, this.secretsFileName);
+        this.secretFieldsSet = new Set(['accessToken', 'githubToken']);
+
+        this._fs = fs;
+        this._fsUtils = fsUtils;
+        this._logger = logger;
+    }
+
+    /**
+     * @param {boolean} ignoreFileNotExists
+     * @param {boolean} ignoreMissingFields
+     * @returns {object|null}
+     */
+    async read(ignoreFileNotExists = false, ignoreMissingFields = false) {
+        if (this._fs.existsSync(this.oldConfigPath)) {
+            let parsedConfig;
+            try {
+                parsedConfig = await this._fsUtils.parseJsonFile(this.oldConfigPath);
+                // Tolerate broken files. We should migrate the old config first
+                //  and then validation will throw an error about missing fields
+                // eslint-disable-next-line no-empty
+            } catch {
+                parsedConfig = {};
+            }
+            await this._migrateOldConfig(parsedConfig);
+            return this._validateStencilConfig(parsedConfig, ignoreMissingFields);
+        }
+
+        const generalConfig = this._fs.existsSync(this.configPath)
+            ? await this._fsUtils.parseJsonFile(this.configPath)
+            : null;
+        const secretsConfig = this._fs.existsSync(this.secretsPath)
+            ? await this._fsUtils.parseJsonFile(this.secretsPath)
+            : null;
+        if (generalConfig || secretsConfig) {
+            const parsedConfig = { ...generalConfig, ...secretsConfig };
+            return this._validateStencilConfig(parsedConfig, ignoreMissingFields);
+        }
+
+        if (ignoreFileNotExists) {
+            return null;
+        }
+
+        throw new Error('Please run'.red + ' $ stencil init'.cyan + ' first.'.red);
+    }
+
+    /**
+     * @param {object} config
+     */
+    async save(config) {
+        const { generalConfig, secretsConfig } = this._splitStencilConfig(config);
+
+        await this._fs.promises.writeFile(this.configPath, JSON.stringify(generalConfig, null, 2));
+        await this._fs.promises.writeFile(this.secretsPath, JSON.stringify(secretsConfig, null, 2));
+    }
+
+    /**
+     * @private
+     * @param {object} config
+     */
+    _splitStencilConfig(config) {
```

Comment:
maybe its time to use private fields? They are supported from node 12
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields

Or do you want to id separately across the project?

Reply:
We still support Node 10 in this project (see https://github.com/bigcommerce/stencil-cli/pull/645#discussion_r492898129) and it doesn’t have support for private fields yet

Reply:
uh, okay, I thought we already dropped the node 10